File: snippets/fsharp/System/GC/Collect/collect4.fs
module collect4

open System
open System.Runtime

let createObjects () =
    Array.init 10000 (fun i -> 
        let s1= "word1"
        let s2 = "word2"
        s1 + " " + s2 )
    |> ignore

createObjects ()
printfn $"Memory allocated before GC: {GC.GetTotalMemory false:N0}"
// <Snippet1>
GCSettings.LargeObjectHeapCompactionMode <- GCLargeObjectHeapCompactionMode.CompactOnce
GC.Collect(2, GCCollectionMode.Forced, true, true)
// </Snippet1>
printfn $"Memory allocated after GC: {GC.GetTotalMemory false:N0}"

File: snippets/csharp/System/GC/Collect/collect4.cs
using System;
using System.Runtime;

public class CollectMemoryExample
{
    public static void Run()
    {
        CreateObjects();
        Console.WriteLine($"Memory allocated before GC: {GC.GetTotalMemory(false):N0}");
        // <Snippet1>
        GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
        GC.Collect(2, GCCollectionMode.Forced, true, true);
        // </Snippet1>
        Console.WriteLine($"Memory allocated after GC: {GC.GetTotalMemory(false):N0}");
    }

    private static void CreateObjects()
    {
        string[] str = new string[10000];
        for (int ctr = 0; ctr < str.Length; ctr++)
        {
            string s1 = "word1";
            string s2 = "word2";
            str[ctr] = s1 + " " + s2;
        }
    }
}
